From 7df87aca022ab6a309a6ee515370686aa3bcc013 Mon Sep 17 00:00:00 2001
From: "Uwe L. Korn" <uwe.korn@quantco.com>
Date: Mon, 14 Dec 2020 13:37:06 +0100
Subject: [PATCH] ARROW-10873: [C++] Apple Silicon is reported as arm64 in
 CMake
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

I'll add a CI entry soon that cross-compiles using `conda`, for now I keep submitting patches when things break for me.

Closes #8893 from xhochy/ARROW-10873

Authored-by: Uwe L. Korn <uwe.korn@quantco.com>
Signed-off-by: Krisztián Szűcs <szucs.krisztian@gmail.com>
---
 cpp/cmake_modules/SetupCxxFlags.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cpp/cmake_modules/SetupCxxFlags.cmake b/cpp/cmake_modules/SetupCxxFlags.cmake
index 402d18f19d6..03d6a4ea34e 100644
--- a/cpp/cmake_modules/SetupCxxFlags.cmake
+++ b/cpp/cmake_modules/SetupCxxFlags.cmake
@@ -24,7 +24,7 @@ include(CheckCXXSourceCompiles)
 message(STATUS "System processor: ${CMAKE_SYSTEM_PROCESSOR}")
 
 if(NOT DEFINED ARROW_CPU_FLAG)
-  if(CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64|ARM64")
+  if(CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64|ARM64|arm64")
     set(ARROW_CPU_FLAG "armv8")
   elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "armv7")
     set(ARROW_CPU_FLAG "armv7")
